src/application/EncryptionInterface.h

```diff
@@ -5,6 +5,8 @@
 #ifndef SRC_ENCRYPTIONINTERFACE_H
 #define SRC_ENCRYPTIONINTERFACE_H
 
+#include "../plugins/encryption/KeyDerivation.h"
```

Comment:
This is a dependency which goes from the application towards the plugin. This should not be the case. Dependencies should only point inwards. To fix this make the parameter `master` only and instead call the derive function inside the implementation of encrypt and decrypt. You could also introduce another abstract class for the key derivation but personally I would prefer the first solution.